File: Arrays/P014.py
def brute(nums):
    n=len(nums)
    longest=seq=0
    for i in range(n-1):
        seq=0
        x=nums[i]
        while x in nums:
            seq+=1
            x+=1
        longest=max(longest,seq)
    return longest

def optimal(nums):
    n=len(nums)
    longest=seq=0
    for i in range(n-1):
        if nums[i]-1 not in nums:
            x=nums[i]
            seq=0
            while x in nums:
                seq+=1
                x+=1
            longest=max(longest,seq)
    return longest

if __name__=='__main__':
    nums=input('Enter numbers: ').split()
    nums=list(map(int,nums))
    print('Brute: ',brute(nums))
    print('Optimal: ',optimal(nums))

File: Stiver_Problems/Arrays/P014.cpp
// P014.cpp - Longest consecutive sequence

#include <bits/stdc++.h>
using namespace std;

int brute(const vector<int>& nums) {
    int n = nums.size();
    int longest = 0;
    for (int i = 0; i < n - 1; i++) {
        int seq = 0;
        int x = nums[i];
        while (find(nums.begin(), nums.end(), x) != nums.end()) {
            seq++;
            x++;
        }
        longest = max(longest, seq);
    }
    return longest;
}

int optimal(const vector<int>& nums) {
    unordered_set<int> s(nums.begin(), nums.end());
    int longest = 0;
    for (int num : nums) {
        if (s.find(num - 1) == s.end()) { // start of a sequence
            int x = num;
            int seq = 0;
            while (s.find(x) != s.end()) {
                seq++;
                x++;
            }
            longest = max(longest, seq);
        }
    }
    return longest;
}

int main() {
    cout << "Enter numbers: ";
    string line;
    getline(cin, line);
    stringstream ss(line);
    vector<int> nums;
    int x;
    while (ss >> x) nums.push_back(x);

    cout << "Brute: " << brute(nums) << "\n";
    cout << "Optimal: " << optimal(nums) << "\n";
    return 0;
}
